### PR TITLE
Basic `ProxyConfigurationManagerTest`

### PR DESCRIPTION
**Pull request by jglick**
Came across #8693 and noticed that the XML serial form of `ProxyConfiguration` was not tested. This is not trivial as can be seen from https://github.com/jenkinsci/jenkins/blob/8b949243a92cc0ff89742be6b509920db0e0a456/core/src/main/java/hudson/ProxyConfiguration.java#L296-L306 https://github.com/jenkinsci/jenkins/blob/8b949243a92cc0ff89742be6b509920db0e0a456/core/src/main/java/hudson/ProxyConfiguration.java#L506-L512

(There is also a getter & setter in `Jenkins` from #3935 which is tested downstream in `configuration-as-code-plugin`: `ProxyConfiguratorTest`.)

### Testing done

Test passes locally.

### Proposed changelog entries

- N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by julieheard — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
